### PR TITLE
Autodocs

### PR DESCRIPTION
**Pull request by tmccombs**
Add github actions workflow to automate api doc generation.

### CODE REVIEW

**Comment by SrodriguezO** [comment]
I thought the `release` and `publish` steps of CI only ran on tag pushes; how come they ran here? Also, are the failures there expected?

**Comment by tmccombs** [comment]
it's weird that they showed up here. I temporarily added a tag to test the doc publishing, but I think the tag may have been a duplicate of a previous tag, so publishing to sonatype failed. It was a while ago, so I may be misremembering.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -80,3 +80,51 @@ jobs:
...
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          path: code
+      - name: Checkout pages
+        uses: actions/checkout@v2
+        with:
+          ref: refs/heads/gh-pages
+          path: pages
+          fetch-depth: 0
```
Comment (SrodriguezO) [inline .github/workflows/ci.yml]: why do you need to fetch the full history of the pages?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -80,3 +80,51 @@ jobs:
...
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+            ~/.coursier
+            target/
+            */target/
+          key: ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('**/*.scala') }}
+          restore-keys: |
+            ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}-
+            ${{ runner.os }}-sbt-
```
Comment (SrodriguezO) [inline .github/workflows/ci.yml]: I feel like I'm missing something with how restore-keys work; from the docs, it sounds like they'll provide a hit if there's any entries in the cache that match this prefix. But that seems like it'd give you false hits? If the main key was missed, that means the Scala source changed. Am I missing something?

TBH this is probably not built enough for a complex cache to be worthwhile. I'd remove the restore-keys form the cache. And in retrospect, I'd remove them from the other jobs' caches too unless you're confident they won't ever cause false hits (and if so, please explain this to me :smile:)

**Review by SrodriguezO — CHANGES_REQUESTED** [review]
LGTM; the deep fetch is mostly a nit, the restore keys I'd like to understand before approving or we can remove them (I don't think they add much)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tmccombs) [reply .github/workflows/ci.yml]: I.... don't remember.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (tmccombs) [reply .github/workflows/ci.yml]: The idea here is that even if the source files are different we can still use the cache for upstream packages.  

You're right though, this won't be built very often. I can remove it.

**Pull request merged**